libs/agno/agno/team/team.py

```diff
@@ -6840,27 +6820,120 @@ def get_messages_for_session(
         if session_id is None:
             raise ValueError("Session ID is required")
 
-        # TODO: Add a db call to get the session summary
         return self.team_session.get_session_summary(session_id=session_id, user_id=user_id)
 
-    def get_user_memories(self, user_id: Optional[str] = None):
+    def get_user_memories(self, user_id: Optional[str] = None) -> Optional[List[UserMemory]]:
         """Get the user memories for the given user ID."""
-        if self.memory is None:
+        if self.memory_manager is None:
             return None
         user_id = user_id if user_id is not None else self.user_id
         if user_id is None:
             user_id = "default"
 
-        if isinstance(self.memory, Memory):
-            return self.memory.get_user_memories(user_id=user_id)
-        raise ValueError(f"Memory type {type(self.memory)} not supported")
+        return self.memory_manager.get_user_memories(user_id=user_id)
+
+    def add_interaction_to_team_context(
+        self, session_id: str, member_name: str, task: str, run_response: Union[RunResponse, TeamRunResponse]
+    ) -> None:
+        if self._team_context is None:
+            self._team_context = {}
+        if session_id not in self._team_context:
+            self._team_context[session_id] = {}
+        self._team_context[session_id] = {
+            "member_name": member_name,
+            "task": task,
+            "response": run_response,
+        }
+        log_debug(f"Updated team context with member name: {member_name}")
+
+    def set_team_context_text(self, session_id: str, text: Union[dict, str]) -> None:
+        if self.team_session_state is None:
+            self.team_session_state = {}
+        if session_id not in self.team_session_state:
+            self.team_session_state[session_id] = {}
+        if isinstance(text, dict):
+            if self.team_session_state[session_id].text is not None:
+                try:
+                    current_context = self.team_session_state[session_id]
+                except Exception:
+                    current_context = {}
+            else:
+                current_context = {}
+            current_context.update(text)
+            self.team_session_state[session_id] = current_context
+        else:
+            # If string then we overwrite the current context
+            self.team_session_state[session_id] = text
+
+    def get_team_context_str(self, session_id: str) -> str:
+        if not self.team_session_state:
+            return ""
+        session_team_context = self.team_session_state.get(session_id, None)
+        if session_team_context:
+            return f"<team context>\n{session_team_context}\n</team context>\n"
+        return ""
+
+    def get_team_member_interactions_str(self, session_id: str) -> str:
+        if not self._team_context:
+            return ""
+        team_member_interactions_str = ""
+        session_team_context = self._team_context.get(session_id, None)
```

Comment:
I know this might be a regression, but I don't think we should do this session wide. Having all of the previous interactions for all members of the session sent to other members is a bit crazy